app/services/user-agent-service/datomstore.js

```diff
@@ -25,81 +25,123 @@ import microtime from 'microtime-fast';
 import { Bookmark } from '../../shared/model';
 import { logger } from '../../shared/logging';
 
-import { SessionEndReason, SessionStartReason, SnippetSize, StarOp, VisitType } from './storage';
+import {
+  SessionEndReason,
+  SessionStartReason,
+  SnippetSize,
+  // StarOp,
+  // VisitType
+} from './storage';
+
+const { lazySeq, vector, parse, getIn } = mori;
+const { DB_ADD, TEMPIDS } = helpers;
+
+const profileSchema = {
+  'page/url': {
+    ':db/cardinality': ':db.cardinality/one',
+    ':db/unique': ':db.unique/identity',
+    ':db/doc': 'A page\'s URL.',
+    ':db.install/_attribute': ':db.part/db',
+  },
+
+  'page/title': {
+    ':db/ident': ':page/title',
+    ':db/cardinality': ':db.cardinality/one',      // We supersede as we see new titles.
+    ':db/doc': 'A page\'s title.',
+    ':db.install/_attribute': ':db.part/db',
+  },
+
+  'event/visit': {
+    ':db/valueType': ':db.type/ref',
+    ':db/cardinality': ':db.cardinality/many',
+    ':db/doc': 'A visit to the page.',
+  },
+
+  'event/star': {
+    ':db/valueType': ':db.type/ref',
+    ':db/cardinality': ':db.cardinality/many',
+    ':db/doc': 'A starring to the page.',
+  },
+
+  // TODO: model sessions.
+  'visit/instant': {
+    ':db/cardinality': ':db.cardinality/one',
+    ':db/doc': 'When the visit occurred.',
+    ':db/index': true,
+    // ':db/valueType': 'db.type/instant',        // Not supported by Datascript.
+  },
+
+  'session/startedFromAncestor': {
+    ':db/valueType': ':db.type/ref',     // To a session.
+    ':db/cardinality': ':db.cardinality/one',
+    ':db/doc': 'The ancestor of a session.',
+  },
+
+  'session/startedInScope': {
```

Comment:
I have a fairly strong preference for `url/style/names` rather than arguing between `camelCaseNames` and `underbar_style_names`.